server/index.js

```diff
@@ -94,7 +94,7 @@ global.LoginServer = require('../.server-dist/loginserver').LoginServer;
 
 global.Ladders = require('./ladders');
 
-global.Chat = require('./chat');
+global.Chat = require('../.server-dist/chat');
```

Comment:
Perhaps export one big object named `Chat` instead of a bunch of little things?

Reply:
I'm not sure if that really achieves much, especially since (afaik) you'd still need to export the types separately. "Exporting a bunch of little things" doesn't really seem bad to me, personally.

Reply:
You wouldn't have to. I believe the correct approach is to export a namespace with the types.

Reply:
The reason it's bad is because it makes the APIs inconsistent. Which means you have to think while importing it, what the correct way to import is, instead of there always being one correct way you can rely upon.

Reply:
maybe putting the utility string functions (html, plural, count, etc) into a static class, à la tools?

Reply:
>The reason it's bad is because it makes the APIs inconsistent.

Don't we only use the way of exporting a big object elsewhere because we need it for the `Object.assign(function, object)` pattern?

Reply:
...no? Chat never even used the `Object.assign(function, object)`. I just did it to make one API nicer. `Users` and `Rooms` were a big object even back when the API was `Users.get()` rather than `Users()`.

Reply:
The `Object.assign(function, object)` pattern can (should) be removed, I just havent gotten around to it yet

Reply:
I feel like `Users` is the exception here, since you need to import it as `require('./users').Users`

Reply:
I made every file work that way after they were converted to TypeScript, though.

Reply:
`Users` may be the exception compared to other `.js` files, but not to other `.ts` files.